protocols/gossipsub/src/behaviour.rs

```diff
@@ -576,15 +571,10 @@ where
         // announce to all peers
         let peer_list = self.peer_topics.keys().cloned().collect::<Vec<_>>();
         if !peer_list.is_empty() {
-            let event = Rpc {
-                messages: Vec::new(),
-                subscriptions: vec![Subscription {
-                    topic_hash: topic_hash.clone(),
-                    action: SubscriptionAction::Unsubscribe,
-                }],
-                control_msgs: Vec::new(),
-            }
-            .into_protobuf();
+            let event = RpcOut::Subscriptions(vec![Subscription {
+                topic_hash: topic_hash.clone(),
+                action: SubscriptionAction::Unsubscribe,
+            }]);
```

Comment:
I would like to remove as many of the nested lists as possible. If we ever need to send multiple subscriptions at once somewhere (here we don't seem to be?), then we can just send multiple events instead.

Reply:
We could even simplify this further by making `RpcOut` have `Subscribe` and `Unsubscribe` variants, then we don't need to pass the `SubscriptionAction` down.

In general, I think it will lead to much easier to understand code if we move towards flat hierarchies here that avoid nested matching and loops.